clients/src/main/java/org/apache/kafka/common/utils/Utils.java

```diff
@@ -331,6 +331,41 @@ public static ByteBuffer wrapNullable(byte[] array) {
         return Arrays.copyOf(src, src.length);
     }
 
+    /**
+     * Compares two character arrays for equality. Two arrays are equal if they have the same length
+     * and all characters at corresponding positions are equal.
+     *
+     * <p>Implementation Note: All characters in the first array are examined to determine equality.
+     * The calculation time depends only on the length of this first character array; it does not
+     * depend on the length of the second character array or the contents of either array.
+     *
+     * @param first the first array to compare
+     * @param second the second array to compare
+     * @return true if the arrays are equal, or false otherwise
+     */
+    public static boolean isEqual(char[] first, char[] second) {
+        if (first == second) {
+            return true;
+        }
+        if (first == null || second == null) {
+            return false;
+        }
+
+        if (second.length == 0) {
+            return first.length == 0;
+        }
+
+        // time-constant comparison that always compares all characters in first array
+        boolean matches = first.length == second.length;
+        for (int i = 0; i < first.length; ++i) {
+            int j = i < second.length ? i : 0;
+            if (first[i] != second[j]) {
+                matches = false;
```

Comment:
We can return `false` on the first mismatch no need to check the rest of the arrays.

Reply:
I believe the goal is to use constant-time comparison to prevent timing attacks, hence the walk through the arrays.

Reply:
I see, my bad. What about renaming the method to make it explicit, something like `isEqualConstantTime()`?